source/language/openpulse.rst

```diff
@@ -0,0 +1,609 @@
+OpenPulse Grammar
+=================
+
+OpenQASM allows users to provide the target system's implementation of quantum operations
+with ``defcal`` blocks . Calibration grammars are open to extension for system implementors. In
+this document, we outline one such grammar, OpenPulse, which maybe selected within a supporting
+compiler through the declaration defcalgrammar "openpulse" <VERSION>.
+
+This grammar is motivated by the original OpenPulse specification, a JSON wire-format for
+pulse-level quantum programs defined in the paper `Qiskit Backend Specifications for OpenQASM
+and OpenPulse Experiment`_:cite:`mckay2018`. The textual format described
+here has several advantages over the original JSON format:
+
+- Improved readability
+- Pulse timing is based on instruction ordering and works with programs containing branching
+  control flow
+- Reusable gate calibrations enable more succinct calibration descriptions.
+- Pulse definitions are declared as a calibration for individual circuit instructions attached to
+  physical qubits enabling the microcoding of gate level operations.
+- Richer ability to compose complex pulses through natural DSP-like operations.
+- Clearly defined relationship between pulses, channels and the phases of compilation to
+  hardware resources.
+- Use of multiple frames on a single channel at the same time (see :ref:`Geometric gate`)
+
+
+Openpulse provides a flexible programming model that should extend to many quantum control schemes
+and hardware. At the core of the OpenPulse grammar are the concepts of ``frames``.
+``frames`` are used to schedule the playing of a ``waveform`` or capturing of data
+via a ``channel`` on the target device. A channel is a software abstraction which allows the
```

Comment:
```suggestion
via a ``channel`` on the target device. A channel is a software abstraction which allows the
```
Is the channel the abstraction or is it the `frame`? I thought the channels were supposed to map pretty closely to HW resources (as stated in the next section). They could still be abstractions, but I thought they were supposed to mirror input and output ports relatively closely.